prime-router/docs/openapi.yml

```diff
@@ -9,6 +9,380 @@ info:
+        - in: query
+          name: dateEnd
+          schema:
+            oneOf:
+              - type: integer
+              - type: string
+          required: false
+          description: (`YYYY-MM-DD` || `NNN` - default `0`) The end date or the number of days back from today to collect results from.
+        - in: query
+          name: year
+          schema:
+            type: integer
+          required: false
+          description: (`YYYY` - default `ALL`) Return results from specific Year.
+        - in: query
+          name: month
+          schema:
+            type: integer
+          required: false
+          description: (`MM` - default `ALL`) Return results from specific Month. Requires `year` being set.
+        - in: query
+          name: organizations
+          schema:
+            oneOf:
+              - type: array
+              - type: integer
+          required: false
+          description: (`{INT, INT, INT}` || `INT` - default `ALL`) Return results from specific Organization or array of Organizations. Admin or member of multiple Organizations required.
+        - in: query
+          name: recipients
+          schema:
+            oneOf:
+              - type: array
+              - type: integer
+          required: false
+          description: (`{INT, INT, INT}` || `INT` - default `ALL`) Return results from specific Recipient or array of Recipients.
+        - in: query
+          name: senders
+          schema:
+            oneOf:
+              - type: array
+              - type: integer
+          required: false
+          description: (`{INT, INT, INT}`  || `INT` - default `ALL`) Return results from specific Sender or array of Senders.
+        - in: query
+          name: warnings
+          schema:
+            type: string
+          required: false
+          description: (`ALL` || `WITH` || `WITHOUT` - default `ALL`) If `WITH` only return results with Warning(s). If `WITHOUT` only results without Warning(s).
+        - in: query
+          name: errors
+          schema:
+            type: string
+          required: false
+          description: (`ALL` || `WITH` || `WITHOUT` - default `ALL`) If `WITH` only return results with Error(s). If `WITHOUT` only results without Error(s).
+        - in: query
+          name: receipts
+          schema:
+            type: string
+          required: false
+          description: (`ALL` || `WITH` || `WITHOUT` - default `ALL`) If `WITH` only return results with Receipt(s). If `WITHOUT` only results without Receipt(s).
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Report'
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+
+  /submissions/{reportID}:
```

Comment:
Note that failed submissions do not have a reportId, but they would have an actionId.  

A query like this could support both failed and successful submissions:
`/submissions/{actionID}   (or maybe /submissions/action/{actionID} or something like that)`
because, given the actionID, you can find the one submitted report associated with it.

However, our current json response only returns the reportId, not the actionId.   So we also need to support this:

`/submissions/{reportID} or maybe /submissions/report/{reportID} or something like that`

... and because each report FKs to a single Action, this can find the action as well.

All this is a long-winded way of saying, we should support queries both on actionIDs and reportIDs.  In both cases they return the json response string.